src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
```

Comment:
*blush* In that case it really doesn’t make sense to do any differently, sorry!

But in fact, there’s even simpler version of the code here, isn’t there? Since the URL is already being stored in the `url_path` buffer (which is where `urlc.lpszUrlPath` points to), what about just querying `INTERNET_OPTION_URL` into that directly and not having to allocate a second buffer:

    InternetQueryOptionA(conn, INTERNET_OPTION_URL, urlc.lpszUrlPath, &urlc.dwUrlPathLength);

? Am I missing something again?